packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -2,28 +2,115 @@ import Sketch from '@sketch-hq/sketch-file-format-ts';
 import produce from 'immer';
 import { AffineTransform, transformRect } from 'noya-geometry';
 import { uuid } from 'noya-renderer';
+import { IndexPath } from 'tree-visit';
 import * as Layers from '../layers';
 import * as Models from '../models';
 import {
-  getBoundingRect,
+  deleteLayers,
   getCurrentPage,
+  getBoundingRect,
   getCurrentPageIndex,
-  getLayerIndexPathsExcludingDescendants,
+  getSymbolsPageIndex,
   getLayerTransformAtIndexPath,
+  getLayerIndexPathsExcludingDescendants,
 } from '../selectors/selectors';
 import { SelectionType, updateSelection } from '../utils/selection';
 import { ApplicationState } from './applicationReducer';
+import { createPage } from './pageReducer';
 
 export type LayerAction =
   | [type: 'deleteLayer', layerId: string | string[]]
   | [type: 'groupLayer', layerId: string | string[], name: string]
   | [type: 'ungroupLayer', layerId: string | string[]]
+  | [type: 'addSymbol', layerId: string | string[], name: string]
   | [
       type: 'selectLayer',
       layerId: string | string[] | undefined,
       selectionType?: SelectionType,
     ];
 
+const getParentLayer = (page: Sketch.AnyLayer, indexPath: IndexPath) =>
+  Layers.access(page, indexPath.slice(0, -1)) as Layers.ParentLayer;
+
+const addSiblingLayer = (
+  page: Sketch.AnyLayer,
+  indexPath: IndexPath,
+  layer: Layers.ChildLayer,
+) => {
+  const parent = getParentLayer(page, indexPath);
+
+  parent.layers.splice(indexPath[indexPath.length - 1], 0, layer);
+};
+
+const getIndexPathsForGroup = (state: ApplicationState, ids: string[]) => {
+  return getLayerIndexPathsExcludingDescendants(
+    state,
+    ids,
+  ).filter((indexPath) =>
+    Layers.isChildLayer(Layers.access(getCurrentPage(state), indexPath)),
+  );
+};
+
+const createGroup = <T extends Sketch.Group | Sketch.SymbolMaster>(
+  model: T,
+  page: Sketch.Page,
+  ids: string[],
+  name: string,
+  indexPaths: IndexPath[],
+): T | undefined => {
+  const boundingRect = getBoundingRect(page, AffineTransform.identity, ids, {
+    clickThroughGroups: true,
+    includeHiddenLayers: true,
+  });
+
+  if (!boundingRect) {
+    console.info('[groupLayer] Selected layers not found');
```

Comment:
Since we log here, we can probably remove the `console.info` call in both places that call `createGroup`